contracts/root/StakeManagerInterface.sol

```diff
@@ -3,11 +3,11 @@ pragma solidity ^0.4.24;
 
 // ERC900
 interface StakeManagerInterface {
-  event Staked(address indexed user, uint256 amount, uint256 total, bytes data);
+  event Staked(address indexed user, address indexed signer, uint256 indexed activatonEpoch, uint256 amount, uint256 total,  bytes data);
   event Unstaked(address indexed user, uint256 amount, uint256 total, bytes data);
 
-  function stake(uint256 amount, bytes data) public;
-  function stakeFor(address user, uint256 amount, bytes data) public;
+  function stake(address unstakeValidator, address signer, uint256 amount,  bytes data) public;
```

Comment:
Do we need `bytes data`?

Reply:
i think for signer we do need `bytes data`

Reply:
@vaibhavchellani signer is a param now.

Reply:
i thought we cannot add in param due to EIP-900 signature matching